system/Helpers/url_helper.php

```diff
@@ -71,16 +82,7 @@ function _get_uri(string $relativePath = '', ?App $config = null): URI
             }
         }
 
-        $url .= $relativePath;
-
-        $uri = new URI($url);
-
-        // Check if the baseURL scheme needs to be coerced into its secure version
-        if ($config->forceGlobalSecureRequests && $uri->getScheme() === 'http') {
-            $uri->setScheme('https');
-        }
-
-        return $uri;
+        return new URI($url . $relativePath);
```

Comment:
What about the scenario when we use `base_url()` and we have set `forceGlobalSecureRequests` to `true`?

Reply:
Thanks. Added a commit to fix it.

I'm not sure this is correct usage, but:
```diff
--- a/app/Config/App.php
+++ b/app/Config/App.php
@@ -130,7 +130,7 @@ class App extends BaseConfig
      * secure, the user will be redirected to a secure version of the page
      * and the HTTP Strict Transport Security header will be set.
      */
-    public bool $forceGlobalSecureRequests = false;
+    public bool $forceGlobalSecureRequests = true;
 
     /**
      * --------------------------------------------------------------------------
diff --git a/app/Controllers/Home.php b/app/Controllers/Home.php
index 7f867e95ff..97f1617d13 100644
--- a/app/Controllers/Home.php
+++ b/app/Controllers/Home.php
@@ -6,6 +6,6 @@ class Home extends BaseController
 {
     public function index()
     {
-        return view('welcome_message');
+        return base_url();
     }
 }
```
```console
$ php public/index.php 
https://localhost:8080/
```

Reply:
Thanks!